web/impact/impact/v1/helpers/profile_helper.py

```diff
@@ -7,58 +32,218 @@
     "prefer not to state": "p",
 }
 
+USER_TYPE_TO_PROFILE_MODEL = {
+    EntrepreneurProfile.user_type: EntrepreneurProfile,
+    ExpertProfile.user_type: ExpertProfile,
+    MemberProfile.user_type: MemberProfile,
+}
+VALID_USER_TYPES = USER_TYPE_TO_PROFILE_MODEL.keys()
+
 VALID_GENDERS = GENDER_TRANSLATIONS.values()
 
-INVALID_GENDER_ERROR = ("Invalid gender: '{}'. Valid values are "
-                        "'f' or 'female', 'm' or 'male', "
-                        "'o' or 'other', and 'p' or 'prefer not to state'")
+VALID_EXPERT_CATEGORIES = [
+    "Executive",
+    "Investor",
+    "Lawyer",
+    "Other",
+]
+
+INVALID_INDUSTRY_ID_ERROR = ("Invalid {field}: "
+                             "Expected valid id for an industry resource")
+INVALID_PROGRAM_FAMILY_ID_ERROR = (
+    "Invalid {field}: Expected valid id for an program family resource")
+
+PHONE_REGEX = re.compile(r'^[0-9x.+() -]+$')
+TWITTER_REGEX = re.compile(r'^\S+$')
+
+EXPERT_ONLY = [ExpertProfile.user_type]
+NON_MEMBER = [ExpertProfile.user_type, EntrepreneurProfile.user_type]
+
+
+def validate_expert_only_boolean(helper, field, value):
+    value = validate_boolean(helper, field, value)
+    return validate_by_user_type(helper, field, value, EXPERT_ONLY)
+
+
+def validate_expert_only_string(helper, field, value):
+    validate_string(helper, field, value)
+    return validate_by_user_type(helper, field, value, EXPERT_ONLY)
+
+
+def validate_non_member_string(helper, field, value):
+    validate_string(helper, field, value)
+    return validate_by_user_type(helper, field, value, NON_MEMBER)
+
+
+def validate_by_user_type(helper, field, value, user_types):
+    user_type = helper.find_user_type()
+    if user_type not in user_types:
+        helper.errors.append(INVALID_CHOICE_ERROR.format(
+                field=field,
+                value=user_type,
+                choices=format_choices(user_types)))
+    return value
+
+
+def validate_personal_website_url(helper, field, value):
+    # This is essentially copied from mc.models.utils in accelerate.
```

Comment:
In the interest of sticking to our [open source standards](https://github.com/masschallenge/standards/blob/master/open_source_repos.md), we should either remove this entire comment block, or make a sanitized edit that doesn't reference MC, Accelerate, or JIRA ticket numbers.